multitenancy/src/main/java/org/axonframework/extensions/multitenancy/components/deadletterqueue/MultiTenantDeadLetterQueue.java

```diff
@@ -0,0 +1,535 @@
+        R res = new TenantWrappedTransactionManager(tenantDescriptor).fetchInTransaction(() -> fetchBlock.apply(
+                getTenantSegment(tenantDescriptor)));
+        return res;
+    }
+
+    private void executeForTenantSegment(TenantDescriptor tenantDescriptor,
+                                         Consumer<SequencedDeadLetterQueue<M>> executeBlock) {
+        new TenantWrappedTransactionManager(tenantDescriptor).fetchInTransaction(() -> {
+            executeBlock.accept(getTenantSegment(tenantDescriptor));
+            return null;
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Registration registerTenant(TenantDescriptor tenantDescriptor) {
+        tenants.add(tenantDescriptor);
+        return () -> {
+            tenants.remove(tenantDescriptor);
+            tenantSegments.remove(tenantDescriptor);
+            return true;
+        };
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Registration registerAndStartTenant(TenantDescriptor tenantDescriptor) {
+        return registerTenant(tenantDescriptor);
+    }
+
+    /**
+     * Return the processing group that this queue is bound to.
+     *
+     * @return The processing group that this queue is bound to.
+     */
+    public String processingGroup() {
+        return processingGroup;
+    }
+
+    /**
+     * Builder class to instantiate a {@link MultiTenantDeadLetterQueue}.
+     *
+     * @param <M> the type of {@link EventMessage} contained in the {@link DeadLetter}
+     */
+    public static class Builder<M extends EventMessage<?>> {
+
+        private TargetTenantResolver<M> targetTenantResolver;
+        private String processingGroup;
+
+        /**
+         * Sets the {@link TargetTenantResolver} used to resolve correct tenant segment based on {@link Message}
+         * message.
+         * <p>
+         * This is a <b>hard requirement</b> and as such should be provided.
+         *
+         * @param targetTenantResolver used to resolve correct tenant segment based on {@link Message} message
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public MultiTenantDeadLetterQueue.Builder<M> targetTenantResolver(
+                TargetTenantResolver<M> targetTenantResolver) {
+            BuilderUtils.assertNonNull(targetTenantResolver, "The TargetTenantResolver is a hard requirement");
+            this.targetTenantResolver = targetTenantResolver;
+            return this;
+        }
+
+        /**
+         * Sets the processing group that this queue is bound to.
+         *
+         * @param processingGroup The processing group that this queue is bound to.
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public MultiTenantDeadLetterQueue.Builder<M> processingGroup(String processingGroup) {
+            this.processingGroup = processingGroup;
```

Comment:
Perhaps my wording was a little off, but I didn't mean to remove the validation for the `processingGroup`.
I would recommend to have the following check here:
```java
BuilderUtils.assertNonEmpty(processingGroup, "Can not set processingGroup to an empty String.");
```

Reply:
it was not mandatory, your comment reminded me of it so I removed it instead. Actually its not used anywhere in the code, it just makes debugging easier